server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -94,10 +94,36 @@ protected void doExecuteForked(Task task, ResolveClusterActionRequest request, A
         assert task instanceof CancellableTask;
         final CancellableTask resolveClusterTask = (CancellableTask) task;
         ClusterState clusterState = clusterService.state();
-        Map<String, OriginalIndices> remoteClusterIndices = remoteClusterService.groupIndices(request.indicesOptions(), request.indices());
-        OriginalIndices localIndices = remoteClusterIndices.remove(RemoteClusterAware.LOCAL_CLUSTER_GROUP_KEY);
 
         Map<String, ResolveClusterInfo> clusterInfoMap = new ConcurrentHashMap<>();
+        Map<String, OriginalIndices> remoteClusterIndices;
+        if (request.clusterInfoOnly()) {
+            if (request.queryingCluster()) {
+                /*
+                 * User does not want to check whether an index expression matches, so we use the "*:dummy: index pattern to
```

Comment:
The comment doesn't seem to match the code - the comment says "dummy" and the code is "*:*".